### PR TITLE
Prepare qiskit-terra 0.23.1

### PR DESCRIPTION
**Pull request by jakelishman**

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -15,9 +15,9 @@ dependencies = [
 
 [[package]]
 name = "ahash"
-version = "0.8.0"
+version = "0.8.3"
```
Comment (mtreinish) [inline Cargo.lock:18]: Is this MSRV compatible?

**Review by mtreinish — COMMENTED** [review]
LGTM, just curious to see if the `cargo update` works against the older MSRV. I remember having issues with `ahash` related to the 1.56.1 MSRV here.

File: Cargo.lock (hunk 1 above)
Comment (jakelishman) [reply Cargo.lock]: Oh, good point - I'll just revert the `cargo update`.  It's safest not to do it in a patch release anyway, I suppose, I've just apparently conditioned myself to do it when updating version numbers.  Done in e5ef2c9ba.

**Review by mtreinish — APPROVED** [review]
LGTM, thanks for the quick update

**Pull request merged**